File: src/main/java/com/github/pagviewer/nativebridge/PagFrameClock.java
package com.github.pagviewer.nativebridge;

public final class PagFrameClock {
    private PagFrameClock() {
    }

    public static int delayMillis(float frameRate) {
        if (frameRate <= 0.0f) {
            return 1;
        }
        return Math.max(1, Math.round(1000.0f / frameRate));
    }

    public static int nextFrame(int currentFrame, int frameCount) {
        if (frameCount <= 0) {
            return 0;
        }
        return (currentFrame + 1) % frameCount;
    }
}

File: src/main/kotlin/com/github/pagviewer/nativebridge/PagFrameClock.kt
package com.github.pagviewer.nativebridge

object PagFrameClock {
    fun delayMillis(frameRate: Float): Int {
        if (frameRate <= 0.0f) {
            return 1
        }
        return maxOf(1, Math.round(1000.0f / frameRate))
    }

    fun nextFrame(currentFrame: Int, frameCount: Int): Int {
        if (frameCount <= 0) {
            return 0
        }
        return (currentFrame + 1) % frameCount
    }
}
